src/datasets/iterable_dataset.py

```diff
@@ -499,19 +565,26 @@ def _iter(self):
             ex_iterable = self._ex_iterable
         yield from ex_iterable
 
+    def _iter_shard(self, shard_idx: int):
+        if self._shuffling:
+            ex_iterable = self._ex_iterable.shuffle_data_sources(self._effective_generator())
+        else:
+            ex_iterable = self._ex_iterable
+        yield from ex_iterable.shard_data_sources(shard_idx)
```

Comment:
This is what is called when iterating in a DataLoader worker. The idea is to iterate only on one shard out of the `self.n_shards` available